src/pghstore/_native.py

```diff
@@ -242,15 +244,15 @@ def parse(string, encoding='utf-8'):
             key = unescape(kq)
         else:
             key = match.group('kb')
-        if isinstance(key, str):
+        if isinstance(key, six.binary_type):
             key = key.decode(encoding)
         vq = match.group('vq')
         if vq:
             value = unescape(vq)
         else:
             vn = match.group('vn')
             value = None if vn else match.group('vb')
-        if isinstance(value, str):
+        if isinstance(value, six.binary_type):
```

Comment:
Good question. I focused more on correctness up-front and not on excessive refactoring. I'll look at that now.